WordPress/gutenberg
# Card: update to g2 implementation

## Description


_This is part of a series of PRs with the final goal of achieving the same result as #32243 in smaller (and easier to review) steps._

Requires #32557 and #32561 to be merged first.

- Replace the current implementation of the `<Card />` set of components with the new g2 implementation (found in [`packages/components/src/ui/card`](https://github.com/WordPress/gutenberg/tree/4857ad58c1241b3d63d21a6880c989b85746c3dc/packages/components/src/card), which is based on a new context system and Emotion's `css` function for the styles.
- Keep all components and their existing APIs unchanged. The only notable changes are:
  -  The `isElevated` prop on the `<Card />` component is being soft-deprecated in favour of a new `elevation` prop (which enables fine-grain control on the card's elevation)
  - The `<CardMedia />` component didn't originally have a g2 implementation and was created directly in this PR (pretty much a 1:1 with the current version)
  - The `<CardDivider />` component didn't originally have a g2 implementation and was created directly in this PR. Although its APIs are the same as the old version, internally it now extends the `<Divider />` component
  - The border color has slightly changed to match the same color used by other g2 components
  - Not really a change, but more of a note that the CSS `padding` associated with the `size` property has been tweaked to match the same spacing as in the old version of the components
- Update unit tests to use snapshots instead of hardcoded classnames

## Testing instructions




- [ ] Project compiles, all tests pass
- [ ] Compare storybook between this branch and production
    - [Current production `Card` Storybook](https://wordpress.github.io/gutenberg/?path=/story/components-card--default)
    - New `Card` Storybook:
        - Pull this PR branch on your machine and run `npm run storybook:dev`
        - Visit [`http://localhost:50240/?path=/story/components-card--default`](http://localhost:50240/?path=/story/components-card--default)
    - The only story that is expected to look different (broken) is the `Components / Card / Media - Horizontally Aligned` 
- [ ] Check old vs new READMEs and make sure that the APIs make sense
- [ ] Make sure that usages of `<Card />` in the repo keep working as before

## Screenshots 

![Screenshot 2021-06-09 at 23 45 09](https://user-images.githubusercontent.com/1083581/121433778-bd0ff780-c97c-11eb-91bb-a2b739d10d4a.png)

## Types of changes





New feature

## Checklist:
